integration-tests/test_connection.py

```diff
@@ -0,0 +1,105 @@
+    """
+    url_test = "End Upload URL Connection Test: SUCCESS"
+    api_test = "End API URL Connection Test: SUCCESS"
+
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    test_connection = insights_client.run("--test-connection")
+    assert url_test in test_connection.stdout
+    assert api_test in test_connection.stdout
+
+
+def test_http_timeout(insights_client):
+    """
+    Test http_timeout config option
+    Set http_timeout to a very low value, run --test-connection and check
+    the set time
+    """
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    insights_client.config.http_timeout = 0.1
+    insights_client.config.save()
+
+    output = insights_client.run("--test-connection", check=False)
+    assert output.returncode == 1
+    assert "read timeout=0.1" in output.stdout
+
+
+def test_noauth_proxy_connection(insights_client, test_config):
+    """
+    Test no-auth proxy option with --test-connection option
+    """
+    url_test = "End Upload URL Connection Test: SUCCESS"
+    api_test = "End API URL Connection Test: SUCCESS"
+
+    insights_client.register()
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    no_auth_proxy: str = (
+        "http://"
+        + test_config.get("noauth_proxy", "host")
+        + ":"
+        + str(test_config.get("noauth_proxy", "port"))
+    )
+    insights_client.config.proxy = no_auth_proxy
+    insights_client.config.save()
+
+    test_connection = insights_client.run("--test-connection")
+    assert url_test in test_connection.stdout
+    assert api_test in test_connection.stdout
+
+
+def test_auth_proxy_connection(insights_client, test_config):
+    """
+    Test auth proxy option with --test-connection
+    """
+    url_test = "End Upload URL Connection Test: SUCCESS"
+    api_test = "End API URL Connection Test: SUCCESS"
+
+    insights_client.register()
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    auth_proxy: str = (
+        "http://"
+        + test_config.get("auth_proxy", "username")
+        + ":"
+        + test_config.get("auth_proxy", "password")
+        + "@"
+        + test_config.get("auth_proxy", "host")
+        + ":"
+        + str(test_config.get("auth_proxy", "port"))
+    )
+    insights_client.config.proxy = auth_proxy
+    insights_client.config.save()
+    test_connection = insights_client.run("--test-connection")
+    assert url_test in test_connection.stdout
+    assert api_test in test_connection.stdout
+
+
+def test_url_connection(insights_client):
+    """
+    Test wrong url option with --test-connection
+    """
+    url_test = "End Upload URL Connection Test: SUCCESS"
+    api_test = "End API URL Connection Test: SUCCESS"
+
+    insights_client.register()
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    insights_client.config.base_url = "cert-api.access.redhat.com:442/r/insights"
```

Comment:
this URL is too similar to existing URLs; I'd suggest to use something less potentially easy to mistake, eg `no-such-insights-url.example.com/something`